### ISSUE TITLE
Restarting Upstart Services Doesn't Re-read the config

### ISSUE THREAD

**Issue by solarkennedy** [garethr/garethr-docker#100]
If upstart is in use (debian), then when asked to restart a docker::run thing, it will notify the service:
https://github.com/garethr/garethr-docker/blob/b60e1c4297de4d87f11b67da197fdeb42bf8ba19/manifests/run.pp#L64

That is good. 

However, if options were changes (lets say you change a published port from 80=>88), Upstart will _not_ take those changes into effect after a restart:
http://upstart.ubuntu.com/faq.html#reload
_only_ after a stop.

In my experience, the best practice is to have configuration parameters in a /etc/default/bla file that can be sourced on restart. That way the /etc/init/bla.conf doesn't ever have to change, and parameters are loaded on each restart.

With this docker module, this is not really super, we end up baking all the configuration of a docker::run  into the upstart config.

Alternatively, the only way I've been able to sanely do this in puppet is a puppet `exec` to stop the service and start it back up when the upstart script changes.

Would you accept a PR for that? Or would you consider it outside the scope? (in some sense this is "not a bug" in upstart land)

TLDR:

Expected: Changing the docker::run config will restart a docker with new config
Actual: The docker is restarted, but not with the new config

Proposed fix: A puppet exec to work around the fact that upstart only re-reads its config upon stop/start.
